services/users/service.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package users
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/shared/i18n"
+	"github.com/mattermost/mattermost-server/v5/shared/mlog"
+	"github.com/mattermost/mattermost-server/v5/store"
+)
+
+type ConfigFn func() *model.Config
+
+type UserService struct {
+	store  store.UserStore
+	config ConfigFn
+}
+
+func New(s store.UserStore, cfgFn ConfigFn) *UserService {
+	return &UserService{
+		store:  s,
+		config: cfgFn,
+	}
+}
+
+// CreateUser creates a user
+func (us *UserService) CreateUser(user *model.User) (*model.User, *model.AppError) {
```

Comment:
Let's use this opportunity to get rid of appError from the user layer, and use it only in the app layer. 0/5 on whether you want to do it now or later.

Reply:
Sure, makes sense to me. In that case do we need to handle the error types here? I mean, we can bypass the errors and let app layer handle the error types and actions regarding to it.

Reply:
Yeah, good point. I think overall we should be doing what we did for the store layer. i.e. categorize user types into `invalidUser`, `internalError`. And then map them to an appError in the app layer. For example, I see here only two distinct error types - `api.user.create_user.accepted_domain.app_error` and `app.user.get_total_users_count.app_error`.

Although I realize this would be some additional work. Totally up to you if you want to do it now or later.